node/core/dispute-coordinator/src/initialized.rs

```diff
@@ -1186,6 +1191,89 @@ impl Initialized {
 
 		Ok(())
 	}
+
+	/// Firstly decrements spam slots for validators who voted on potential spam
+	/// candidates that are newly backed or included, and therefore no longer
+	/// potential spam.
+	/// 
+	/// Secondly queues participation if needed for freshly backed or included 
+	/// candidates. Backed candidates are added to best_effort queue while 
+	/// included are added to priority queue.
+	async fn handle_backed_included_events<Context>(
+		&mut self, 
+		ctx: &mut Context, 
+		overlay_db: &mut OverlayedBackend<'_, impl Backend>, 
+		now: u64,
+		candidate_events: Vec<CandidateEvent>,
+	)
+	{
+		let session = self.rolling_session_window.latest_session();
+		// Populate events contents, which allow us to queue participation without
+		// duplicated code
```

Comment:
No need for two loops though or the vec. let event_content = match event {...}; ...